java/developer-guide/working-with-outlook-items/working-with-recurrences/_index.md

```diff
@@ -0,0 +1,527 @@
+
+MapiTask task = new MapiTask("This is test task", "Sample Body", startDate, dueDate);
+task.setState(MapiTaskState.NotAssigned);
+
+// Set the weekly recurrence
+MapiCalendarWeeklyRecurrencePattern rec = new MapiCalendarWeeklyRecurrencePattern();
+rec.setEndType(MapiCalendarRecurrenceEndType.EndAfterNOccurrences);
+rec.setPatternType(MapiCalendarRecurrencePatternType.Week);
+rec.setPeriod(1);
+rec.setWeekStartDay(DayOfWeek.Sunday);
+rec.setDayOfWeek(MapiCalendarDayOfWeek.Friday);
+rec.setOccurrenceCount(getOccurrenceCount(startDate, endByDate, "FREQ=WEEKLY;BYDAY=FR"));
+
+if (rec.getOccurrenceCount() == 0)
+{
+    rec.setOccurrenceCount(1);
+}
+
+task.setRecurrence(rec);
+task.save(dataDir + "Weekly_out.msg", TaskSaveFormat.Msg);
+~~~
+
+
+
+Following function can be used to calculate the number of events between the two dates:
+
+
+
+~~~Java
+private static long getOccurrenceCount(Date start, Date endBy, String rrule)
+{
+    DateFormat formatter = new SimpleDateFormat("yyyyMMdd");
+    CalendarRecurrence pattern = new CalendarRecurrence("DTSTART:" + formatter.format(start) + "\r\nRRULE:" + rrule);
+    DateCollection dates = pattern.generateOccurrences(start, endBy);
+    return dates.size();
+}
+~~~
+#### **Selecting multiple days in a week**
+The following code snippet shows you how to select multiple days in a week.
+
+
+
+~~~Java
+// For complete examples and data files, please go to https://github.com/aspose-email/Aspose.Email-for-Java
+// Set the weekly recurrence
+MapiCalendarWeeklyRecurrencePattern rec = new MapiCalendarWeeklyRecurrencePattern();
+
+rec.setEndType(MapiCalendarRecurrenceEndType.EndAfterNOccurrences);
+rec.setPatternType(MapiCalendarRecurrencePatternType.Week);
+rec.setPeriod(1);
+rec.setWeekStartDay(DayOfWeek.Sunday);
+rec.setDayOfWeek(MapiCalendarDayOfWeek.Friday | MapiCalendarDayOfWeek.Monday);
+rec.setOccurrenceCount(getOccurrenceCount(startDate, endByDate, "FREQ=WEEKLY;BYDAY=FR,MO"));
+~~~
+#### **Selecting multiple days in a week and setting intervals**
+The following code snippet shows you how to Selecting multiple days in a week and setting intervals.
+
+
+
+~~~Java
+// For complete examples and data files, please go to https://github.com/aspose-email/Aspose.Email-for-Java
+// Set the weekly recurrence
+MapiCalendarWeeklyRecurrencePattern rec = new MapiCalendarWeeklyRecurrencePattern();
+rec.setEndType(MapiCalendarRecurrenceEndType.EndAfterNOccurrences);
+rec.setPatternType(MapiCalendarRecurrencePatternType.Week);
+rec.setPeriod(2);
+rec.setWeekStartDay(DayOfWeek.Sunday);
+rec.setDayOfWeek(MapiCalendarDayOfWeek.Friday | MapiCalendarDayOfWeek.Monday);
+rec.setOccurrenceCount(getOccurrenceCount(startDate, endByDate, "FREQ=WEEKLY;BYDAY=FR,MO;INTERVAL=2"));
+~~~
+### **Weekly Recurrences: EndAfterDate Recurrance Type**
+"End By" option in the Mapi Task is achieved by setting the OccurrenceCount property calculated by the GetOccurrenceCount() function. This function takes start date , end date and RRULE string.
```

Comment:
`"End By" option in the Mapi Task is achieved by setting the *OccurrenceCount* property calculated by the **GetOccurrenceCount()** function. This function takes start date, end date and RRULE string.`